drivers/flash/Kconfig.nor

```diff
@@ -0,0 +1,105 @@
+#
+# Copyright (c) 2018 Savoir-Faire Linux.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+
+menuconfig SPI_NOR
+	bool
+	prompt "SPI NOR Flash"
+	select FLASH_HAS_DRIVER_ENABLED
+	depends on SPI && FLASH
+
+if SPI_NOR
+
+config SPI_NOR_INIT_PRIORITY
+	int
+	default 80
+	help
+	  Device driver initialization priority.
+	  Device is connected to SPI bus, it has to
+	  be initialized after SPI driver.
+
+config SPI_NOR_GPIO_SPI_CS
+	bool "Manage SPI CS through a GPIO pin"
+	default n
+	help
+	This option is useful if one needs to manage SPI CS through a GPIO
+	pin to by-pass the SPI controller's CS logic.
+
+config SPI_NOR_GPIO_SPI_CS_DRV_NAME
+	string "GPIO driver's name to use to drive SPI CS through"
+	default ""
+	depends on SPI_NOR_GPIO_SPI_CS
+	help
+	This option is mandatory to set which GPIO controller to use in order
+	to actually emulate the SPI CS.
+
+config SPI_NOR_GPIO_SPI_CS_PIN
+	int "GPIO PIN to use to drive SPI CS through"
+	default 0
+	depends on SPI_NOR_GPIO_SPI_CS
+	help
+	This option is mandatory to set which GPIO pin to use in order
+	to actually emulate the SPI CS.
+
+config SPI_NOR_GPIO_SPI_CS_WAIT_DELAY
+	int "Delay time in us"
+	default 0
+	depends on SPI_NOR_GPIO_SPI_CS
+	help
+	This is the wait delay (in us) to allow for CS switching to take effect
+
+config SPI_NOR_JEDEC_ID
+	hex "Unique Id to identify SPI flash"
+	default 0
+	help
+	This is a unique id to identify SPI flash
+
+config SPI_NOR_PAGE_SIZE
+	int "Page size of SPI flash"
+	default 0
+	help
+	This option specifies page size of SPI flash
+
+config SPI_NOR_SECTOR_SIZE
+	int "Sector size of SPI flash"
+	default 0
+	help
+	This option specifies sector size of SPI flash
+
+config SPI_NOR_SECTORS
+	int "Number of sectors in SPI flash"
+	default 0
+	help
+	This option specifies number of sector present
+	in SPI flash
+
+config SPI_NOR_BLOCK_SIZE
+	int "Block size of SPI flash"
+	default 0
+	help
+	This option specifies block size of SPI flash
+
+choice SPI_NOR_BLOCK_ERASE_SIZE
+prompt "Select Block size for erasing"
+
+config SPI_NOR_BLOCK_ERASE_32K
+	bool "Select 32K block erase"
+	help
+	This Enables 32K Block erase
+
+config SPI_NOR_BLOCK_ERASE_64K
+	bool "Select 64K block erase"
+	help
+	This Enables 64K Block erase
+
+endchoice
+
+config SPI_NOR_BLOCK_ERASE_SIZE
+	int "Block size for erasing SPI flash"
```

Comment:
Just a remark in case you would like that the user can only select values through the choice above remove the text after ```int``` and it will not show up in menu config

Reply:
done.